src/components/app/ProfileDeleteButton.js

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import React, { PureComponent } from 'react';
+import classNames from 'classnames';
+
+import { ButtonWithPanel } from 'firefox-profiler/components/shared/ButtonWithPanel';
+
+import { deleteProfileData } from 'firefox-profiler/app-logic/published-profiles-store';
+import { deleteProfileOnServer } from 'firefox-profiler/profile-logic/profile-store';
+
+import './ProfileDeleteButton.css';
+
+type Props = {|
+  +profileName: string,
+  +smallProfileName: string,
+  +jwtToken: string,
+  +profileToken: string,
+  +buttonClassName?: string,
+  +onOpenConfirmDialog?: () => mixed,
+  +onCloseConfirmDialog?: () => mixed,
+  +onCloseSuccessMessage?: () => mixed,
+|};
+
+type State = {|
+  +status: 'idle' | 'working' | 'just-deleted' | 'deleted',
+  +error: Error | null,
+|};
+
+export class ProfileDeleteButton extends PureComponent<Props, State> {
+  state = { error: null, status: 'idle' };
+  _componentDeleteButtonRef = React.createRef<ButtonWithPanel>();
+
+  onConfirmDeletion = async () => {
+    const { profileToken, jwtToken } = this.props;
+
+    this.setState({ status: 'working' });
+    try {
+      if (!jwtToken) {
+        throw new Error(
+          `We have no JWT token for this profile, so we can't delete it. This shouldn't happen.`
+        );
+      }
+      await deleteProfileOnServer({ profileToken, jwtToken });
+      await deleteProfileData(profileToken);
```

Comment:
I think this was in the other patch as well, but what happens when `await deleteProfileOnServer` is successful, but then `await deleteProfileData` fails? Is the user left in a broken state with a zombie profile listing that won't go away? Maybe we should specifically handle this case. For instance, if `deleteProfileOnServer` is a 404, we should remove the local entry (Although, much more likely, we already deleted it somewhere else, and we never updated the UI)

Reply:
think that `await deleteProfileData` should never fail in the wild. DB errors would appear earlier (when listing profiles).
If `deleteProfileOnServer` is a 404 we should get an error before even calling `deleteProfileData` :-)

And indeed I was thinking that a follow-up could be that we have an action "remove this entry anyway" with the error message. I can file this and maybe this could be a good first issue even.